File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Scanner;
import java.util.Stack;
import java.util.StringTokenizer;

public class Main {
    
    public static void main(String[] args) throws IOException{
    
        Stack<Integer> st=new Stack<>();
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
      
        int num=Integer.parseInt(br.readLine());
        for(int i=0;i<num;i++){
            StringTokenizer str = new StringTokenizer(br.readLine());
            String m=str.nextToken();
             switch(m){
                case "push":
                st.push(Integer.parseInt(str.nextToken()));
                break;
                case "pop":
                {
                    if(st.isEmpty())
                    System.out.println("-1");
                    else 
                    System.out.println(st.pop());
                }break;
                case "top":
                { if(st.isEmpty())
                System.out.println("-1");
                else
                System.out.println(st.peek());
                break;}
                case "size":
                System.out.println(st.size());
                break;
                case "empty":
                {
                    if(st.isEmpty())
                    System.out.println("1");
                    else
                    System.out.println("0");
                }

             }   
        }

        

    }
   
}

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <vector>
#include <stack>

using namespace std;

int main() {
    stack<int> st;
    int n;
    string temp = "";
    cin >> n;
    
    for (int i = 0; i < n; i++)
    {
        cin >> temp;
        if (temp == "push")
        {
            int a;
            cin >> a;
            st.push(a);
        }
        else if (temp == "empty")
        {
            if (st.empty())
            {
                cout << 1 << "\n";
            }
            else
            {
                cout << 0 << "\n";
            }
        }
        else if (temp == "pop") {
            if (st.empty())
            {
                cout << -1 << "\n";
            }
            else
            {
                cout << st.top() << "\n";
                st.pop();
            }
        }
        else if (temp == "size")
        {
            cout << st.size() << "\n";
        }
        else if (temp == "top")
        {
            if (st.empty())
            {
                cout << -1 << "\n";
            }
            else
            {
                cout << st.top() << "\n";
            }
        }
        
    }

}
